GliaWidgets/SecureConversations/ChatTranscript/SecureConversations.TranscriptModel+GVA.swift

```diff
@@ -0,0 +1,126 @@
+import Foundation
+
+private extension String {
+    static let gvaOptionUrlTargetModal = "modal"
+    static let gvaOptionUrlTargetSelf = "self"
+}
+
+extension SecureConversations.TranscriptModel {
+    func quickReplyOption(_ gvaOption: GvaOption) -> QuickReplyButtonCell.Props {
+        let action = Cmd { [weak self] in
+            self?.gvaOptionAction(for: gvaOption)()
+            self?.action?(.quickReplyPropsUpdated(.hidden))
+        }
+        return .init(
+            title: gvaOption.text,
+            action: action
+        )
+    }
+
+    func gvaOptionAction(for option: GvaOption) -> Cmd {
+        // If `option.destinationPdBroadcastEvent` is specified,
+        // this is broadcast event button, which is not supported
+        // on mobile. So an alert should be shown.
+        // If option contains `url`, then it's URL Button,
+        // otherwise it's Postback Button and option should be sent
+        // to the server as `SingleChoiceOption`
+        if option.destinationPdBroadcastEvent != nil {
+            return broadcastEventButtonAction()
+        } else if let urlString = option.url,
+                  let url = URL(string: urlString) {
+            return urlButtonAction(url: url, urlTarget: option.urlTarget)
+        } else {
+            return postbackButtonAction(for: option)
+        }
+    }
+
+    func postbackButtonAction(for option: GvaOption) -> Cmd {
+        .init { [weak self] in
+            self?.sendGvaOption(option)
+        }
+    }
+}
+
+private extension SecureConversations.TranscriptModel {
+    func urlButtonAction(url: URL, urlTarget: String?) -> Cmd {
+        .init { [weak self] in
+            guard let self else { return }
+
+            let openUrl = { [weak self] url in
+                guard let self = self else { return }
+                guard self.environment.uiApplication.canOpenURL(url) else { return }
+                self.environment.uiApplication.open(url)
+            }
+
+            switch url.scheme?.lowercased() {
+            case URLScheme.tel.rawValue,
+                URLScheme.mailto.rawValue,
+                URLScheme.http.rawValue,
+                URLScheme.https.rawValue:
+                // "tel" ,"mailto" and "http(s)"-based links should be opened by UIApplication
+                openUrl(url)
+
+            default:
+                if urlTarget == .gvaOptionUrlTargetModal ||
+                    urlTarget == .gvaOptionUrlTargetSelf {
+                    // if GvaOption.urlTarget is "modal" or "self", then button url is deeplink
+                    // and should be opened by UIApplication, to provide integrator
+                    // an ability to handle deeplinks they configured.
+                    openUrl(url)
+                } else {
+                    return
+                }
+            }
```

Comment:
Please correct me if I missed something, however, it looks like do the same in both cases...
```swift
            switch url.scheme?.lowercased() {
            case URLScheme.tel.rawValue,
                URLScheme.mailto.rawValue,
                URLScheme.http.rawValue,
                URLScheme.https.rawValue:
                // "tel" ,"mailto" and "http(s)"-based links should be opened by UIApplication
                openUrl(url)                       👈

            default:
                if urlTarget == .gvaOptionUrlTargetModal ||
                    urlTarget == .gvaOptionUrlTargetSelf {
                    // if GvaOption.urlTarget is "modal" or "self", then button url is deeplink
                    // and should be opened by UIApplication, to provide integrator
                    // an ability to handle deeplinks they configured.
                    openUrl(url)                    👈
                } else {
                    return
                }
            }
```

would it make sense to remove this logic and just open link after `canOpen` it?